packages/wonder-blocks-layout/src/components/media-layout.tsx

```diff
@@ -239,14 +234,22 @@ class MediaLayoutInternal extends React.Component<CombinedProps, State> {
         // the current MediaSpec.
         const mediaSize =
             overrideSize ||
-            (this.isServerSide() && ssrSize) ||
+            (initialRender && ssrSize) ||
             this.getCurrentSize(mediaSpec);
 
         // Generate a mock stylesheet
         const styles = this.getMockStyleSheet(mediaSize);
 
         return children({mediaSize, mediaSpec, styles});
     }
+
+    render() {
+        return (
+            <WithSSRPlaceholder placeholder={() => this.renderContent(true)}>
+                {() => this.renderContent(this.isUnsupportedEnvironment())}
+            </WithSSRPlaceholder>
+        );
+    }
```

Comment:
**note:** This now ensures we'll hydrate correctly. The initial "placeholder" render will do the old server-side behavior, and then the subsequent "children" render will do the old behavior based on window and matchMedia availability.